### PR TITLE
[#92191452] setup plugin system

### PR DESCRIPTION
**Pull request by kelvintaywl**
This PR tries to set up Goship such that we can customize the projects list page (home page) to include more details / actionable columns for each project through plugins. 

Also, this PR grabs all stylesheets and javascripts from the respective `static/css`, `static/js` folders and inserts them into the templates (I have moved the custom stylings to a `static/css/styles.css` file)

### about plugins

An example helloworld plugin is included in this PR. when importing that plugin on `goship.go`, we can see that it is pretty configurable. 

||
| --- |
| screen shot 2015-04-22 at 5 46 03 pm |

More details can be found on the [README](), but in general, every project has PluginColumns which are customizable. A plugin can add additional columns to the project and render it's own logic / functionality.

#### notes
A plugin is just required to implement a `Apply` method while the Column instance is required to implement the `RenderHeader` and `RenderDetail` methods.

### TODO
- [ ] add tests

### CODE REVIEW

**Comment by hayate** [comment]
:+1: 
LGTM

File: plugins/README.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+```go
+// in goship.go
+package main
+
+import (
+	
+	...
+	...
+
+	helloworld "github.com/gengo/goship/plugins/helloworld"
```
Comment (ramen74) [inline plugins/README.md]: Perhaps can find a better way to do the import then patching the main goship file.

**Comment by ramen74** [comment]
LGTM

**Comment by kelvintaywl** [comment]
@trinchan @ramen74 can we merge this now then? :bow:

File: plugins/README.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+```go
+// in plugins/plugins.go
+package plugins
+
+// Import plugin packages here
+import _ "github.com/gengo/goship/plugins/travis"
+
+```
+
+With this, when Goship is run, we should see the `RenderDetail()` and `RenderHeader()` method of our HelloWorld plugin displaying on the home page!
```
Comment (trinchan) [inline plugins/README.md]: HelloWorld -> Travis?

File: plugins/README.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+![helloworld plugin example](http://i.imgur.com/0r0yZGI.png)
```
Comment (trinchan) [inline plugins/README.md]: We can host this screenshot on github

**Pull request merged**